### PR TITLE
Remove deprecated script APIs

### PR DESCRIPTION
**Pull request by s1monw**
The script APIs have been deprecated long ago we can now remove them.
This commit still keeps the parsing code since it might be used in a
query that is still stuck in transaction log. This issue should be discussed
elsewhere.

### CODE REVIEW

**Comment by kimchy** [comment]
LGTM

File: core/src/main/java/org/elasticsearch/script/Script.java [hunk 1]
```diff
@@ -56,26 +56,66 @@
...
+     * @param script
+     *            The cache key of the script to be compiled/executed. For
+     *            inline scripts this is the actual script source code. For
+     *            indexed scripts this is the id used in the request. For on
+     *            file scripts this is the file name.
+     * @param type
+     *            The type of script -- dynamic, indexed, or file.
+     */
+    public Script(String script, ScriptType type) {
+        this(script, ScriptType.INLINE, null, null);
```
Comment (rjernst) [inline core/src/main/java/org/elasticsearch/script/Script.java]: Do we really need all these additional ctors? This seems to be a bug: `type` is not used here, but instead INLINE is always passed?

**Comment by s1monw** [comment]
@rjernst I pushed a new commit

**Comment by rjernst** [comment]
LGTM

**Pull request merged**